research/object_detection/dataset_tools/create_ava_tf_record.py

```diff
@@ -0,0 +1,439 @@
+              'region/bbox/xmin':
+                  feature_list_feature(total_xmins),
+              'region/bbox/xmax':
+                  feature_list_feature(total_xmaxs),
+              'region/bbox/ymin':
+                  feature_list_feature(total_ymins),
+              'region/bbox/ymax':
+                  feature_list_feature(total_ymaxs),
+              'region/label/index':
+                  feature_list_feature(total_labels),
+              'region/label/string':
+                  feature_list_feature(total_label_strings),
+              'region/label/confidence':
+                  feature_list_feature(total_confidences), #all ones
+              'region/is_annotated':
+                  feature_list_feature(total_is_annotated) #all ones
+          }
+
+          if len(total_xmins) > 0:
+            yield tf.train.SequenceExample(
+                context=tf.train.Features(feature=context_feature_dict),
+                feature_lists=tf.train.FeatureLists(
+                    feature_list=sequence_feature_dict))
+
+          #Move middle_time_frame, skipping excluded frames
+          frames_mv = 0
+          frames_excluded_count = 0
+          while (frames_mv < hop_between_sequences + frames_excluded_count
+                 and middle_frame_time + frames_mv < POSSIBLE_TIMESTAMPS[-1]):
+            frames_mv += 1
+            if (media_id, windowed_timestamp + frames_mv) in frame_excluded:
+              frames_excluded_count += 1
+          middle_frame_time += frames_mv
+
+        cur_vid.release()
+
+  def _download_data(self, download_labels_for_map):
+    """Downloads and extracts data if not already available."""
+    if sys.version_info >= (3, 0):
+      urlretrieve = urllib.request.urlretrieve
+    else:
+      urlretrieve = urllib.request.urlretrieve
+    logging.info("Creating data directory.")
+    tf.io.gfile.makedirs(self.path_to_data_download)
+    logging.info("Downloading annotations.")
+    paths = {}
+    if download_labels_for_map:
+      zip_path = os.path.join(self.path_to_data_download,
+                              ANNOTATION_URL.split("/")[-1])
+      urlretrieve(ANNOTATION_URL, zip_path)
+      with zipfile.ZipFile(zip_path, 'r') as zip_ref:
+          zip_ref.extractall(self.path_to_data_download)
+      for split in ["train", "test", "val"]:
+        csv_path = os.path.join(self.path_to_data_download,
+                                "ava_%s_v2.2.csv" % split)
+        excl_name = "ava_%s_excluded_timestamps_v2.2.csv" % split
+        excluded_csv_path = os.path.join(self.path_to_data_download, excl_name)
+        SPLITS[split]["csv"] = csv_path
+        SPLITS[split]["excluded-csv"] = excluded_csv_path
+        paths[split] = (csv_path, excluded_csv_path)
+
+    label_map = self.get_label_map(os.path.join(self.path_to_data_download,
+                                                "ava_action_list_v2.2.pbtxt"))
+    return paths, label_map
+
+  def get_label_map(self, path):
```

Comment:
use label_map_util.py function to parse label map https://github.com/tensorflow/models/blob/451906e4e82f19712455066c1b27e2a6ba71b1dd/research/object_detection/utils/label_map_util.py#L166

Reply:
The AVA label map is in a different format, unfortunately can't use it.